contracts/libs/Types.sol

```diff
@@ -51,12 +52,34 @@ library Types {
         Usage batchType;
     }
 
+    struct MMCommitment {
+        bytes32 stateRoot;
+        bytes32 accountRoot;
+        bytes32 txHashCommitment;
```

Comment:
If we can use txs here instead of txHashCommitment, we can save some other inputs and checks. Since we committed the whole MMCommitment, no point to hash txs here anymore.

Reply:
Done!